docs/developer/README.md

```diff
@@ -0,0 +1,30 @@
+# Developer Documentation
+
+As a developer, I want to ...
+- ... extend the existing codebase.
+- ... write new extensions.
+- ... work on the service provider interface (SPI).
+- ... improve core implementations.
+- ... have clear requirements.
+- ... add documentation to new and existing code.
+- ... stick to existing coding styleguides.
+- ... get an overview of past decision records.
+- ... improve software tests.
+
+As a developer, I do not want to ...
+- ... bring in new requirements.
+- ... deep-dive into use cases.
+
+## Important links
+
+- [Command Queue](command-queue.md)
+- [Dependency Resolution](dependency_resolution.md)
+- [Data Plane Framework](data-plane-framework/README.md)
+- [Logging](logging.md)
+- [Metrics](metrics.md)
+- [Releases](releases.md)
+- [Testing](testing.md)
+
+> All implementations have to follow existing design principles and architectural patterns that are provided as 
```

Comment:
I would also link the style guide and architectural principles here. can't be too sure...